### PR TITLE
Feature/add proxy settings to nexus & jenkins

### LINKED ISSUE #637: Support for http proxy for Nexus and Jenkins master
**Is your feature request related to a problem?**
In a corp environment there is usually a proxy to filter inet traffic. While all other components already support a proxy - Nexus and Jenkins master dont - which means no updates, no market place

**Describe the solution you'd like**
Full support for http proxy

### PR DESCRIPTION
**Pull request by clemensutschig**
closes #637

### CODE REVIEW

File: nexus/configure.sh [hunk 1]
```diff
@@ -228,7 +238,11 @@ echo_info "Install Blob Stores"
 runJsonScript "createBlobStores"
 
 echo_info "Configure proxy if applicable"
-runJsonScript "createProxySettings"
+sed "s|@http_proxy@|${HTTP_PROXY}|g" json/createProxySettings.json > json/createProxySettingsWithProxy.json
+sed -i "s|@https_proxy@|${HTTPS_PROXY}|g" json/createProxySettingsWithProxy.json
+sed -i "s|@no_proxy@|${NO_PROXY}|g" json/createProxySettingsWithProxy.json
+cat json/createProxySettingsWithProxy.json
+runJsonScript "createProxySettings" "-d @json/createProxySettingsWithProxy.json"
```
Comment (michaelsauter) [inline nexus/configure.sh:246]: Could you delete the file as below? `rm json/developer-user-with-password.json`?

File: nexus/json/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+createProxySettingsWithProxy.json
```
Comment (michaelsauter) [inline nexus/json/.gitignore:1]: Good call, can you add the `developer-user-with-password.json` file as well?

**Comment by clemensutschig** [comment]
@michaelsauter  - I have never seen such a crappy groovy implementation .. neither String.contains nor String [].length works .. it's soo frustrating .. and NO log ...

**Comment by clemensutschig** [comment]
Ok - there is something really weird - 2 ifs in the same script not allowed, which is why i used a ternary operator, and split the settings ... crazy!

**Comment by michaelsauter** [comment]
@clemensutschig I had to make a few adjustments, as you were executing not the changed file, but sending the changed file as params. I restructured it a bit in 31a23b0, and ran it locally, see screenshot below:

Screenshot 2020-06-30 at 16 29 14

**Comment by clemensutschig** [comment]
@michaelsauter  - NICE .. thanks a ton.. approve and I merge :)

**Pull request merged**